server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -220,7 +246,21 @@ private <Params extends PersistentTaskParams> void doStartTask(
                     + "], removing from persistent tasks",
                 e
             );
-            notifyMasterOfFailedTask(taskInProgress, e);
```

Comment:
Why you choose placeholder approach? Can it be a bug in how master handles task failures?

Reply:
No, it's just that we assume here that the `notifyMasterOfFailedTask` takes effect immediately and that isn't true always (see linked case). We need to record the fact that we processed this task creation so that we don't keep on retrying.